drivers/overlay/ov_network.go

```diff
@@ -178,17 +175,88 @@ func (n *network) destroySandbox() {
 				}
 			}
 		}
+
+		if hostMode {
+			if err := removeNetworkChain(n.id[:12]); err != nil {
+				logrus.Warnf("could not remove network chain: %v", err)
+			}
+		}
+
 		sbox.Destroy()
 		n.setSandbox(nil)
 	}
 }
 
-func (n *network) initSubnetSandbox(s *subnet) error {
-	// create a bridge and vxlan device for this subnet and move it to the sandbox
-	brName, err := netutils.GenerateIfaceName("bridge", 7)
+func setHostMode() {
+	if os.Getenv("_OVERLAY_HOST_MODE") != "" {
+		hostMode = true
+		return
+	}
+
+	err := createVxlan("testvxlan", 1)
 	if err != nil {
-		return err
+		logrus.Errorf("Failed to create testvxlan interface: %v", err)
+		return
 	}
+
+	path := "/proc/self/ns/net"
+	f, err := os.OpenFile(path, os.O_RDONLY, 0)
+	if err != nil {
+		logrus.Errorf("Failed to open path %s for network namespace for setting host mode: %v", path, err)
+		return
+	}
+	defer f.Close()
+
+	nsFD := f.Fd()
+
+	iface, err := netlink.LinkByName("testvxlan")
+	if err != nil {
+		logrus.Errorf("Failed to get link testvxlan: %v", err)
+		return
+	}
+
+	// If we are not able to move the vxlan interface to a namespace
+	// then fallback to host mode
+	if err := netlink.LinkSetNsFd(iface, int(nsFD)); err != nil {
+		hostMode = true
+	}
```

Comment:
Yes, it fails.

The disadvantage of kernel version check is that it introduces static assumption on which kernel version this feature exists but it will work the wrong way in the following scenarios:
* Some one backported this enhancement in a previous kernel version
* Even if the kernel version is correct, moving to namespace may not work in all kernel versions i.e in some other arch, or somebody is using a heavily customized kernel

So checking for the actual functionality is the most robust way.